libnmstate/nm/device.py

```diff
@@ -135,24 +135,33 @@ def _safe_activate_async(dev, connection_id):
 
 
 def _active_connection_callback(src_object, result, user_data):
-    mainloop, nmdev, cancellable = user_data
+    mainloop, nmdev, connection_id, cancellable = user_data
     mainloop.drop_cancellable(cancellable)
     try:
         nm_act_con = src_object.activate_connection_finish(result)
     except Exception as e:
+        if nmdev:
```

Comment:
Could you please move this new code to a helper function that returns the two values?

Reply:
done